tianshou/data/collector.py

```diff
@@ -177,43 +172,44 @@ def reset_buffer(self, keep_statistics: bool = False) -> None:
         """Reset the data buffer."""
         self.buffer.reset(keep_statistics=keep_statistics)
 
-    def reset_env(self, gym_reset_kwargs: dict[str, Any] | None = None) -> None:
+    def reset_env(
+        self,
+        gym_reset_kwargs: dict[str, Any] | None = None,
+    ) -> tuple[np.ndarray, dict | list[dict]]:
         """Reset all of the environments."""
         gym_reset_kwargs = gym_reset_kwargs if gym_reset_kwargs else {}
         obs, info = self.env.reset(**gym_reset_kwargs)
-        if self.preprocess_fn:
-            processed_data = self.preprocess_fn(obs=obs, info=info, env_id=np.arange(self.env_num))
-            obs = processed_data.get("obs", obs)
-            info = processed_data.get("info", info)
-        self.data.info = info  # type: ignore
-        self.data.obs = obs
-
-    def _reset_state(self, id: int | list[int]) -> None:
-        """Reset the hidden state: self.data.state[id]."""
-        if hasattr(self.data.policy, "hidden_state"):
-            state = self.data.policy.hidden_state  # it is a reference
+        return obs, info
+
+    def _reset_state(
```

Comment:
this function should be called _reset_hidden_state and be defined outside of the collector (note that the IDE should have warned you that the method can be static). The input should just be called `rollout_batch`, there is no need for the first `if hasattr` (that should happen in downstream code), and the return should be `None`.

Note that this modifies the batch inplace (side effect), and such operations should usually not return anything.

The parameter `id` deserves an explanation